pkg/envoy/envoy.go

```diff
@@ -277,24 +305,18 @@ func (e *Envoy) accessLogger(conn *net.UnixConn) {
 // StopEnvoy kills the Envoy process started with StartEnvoy. The gRPC API streams are terminated
 // first.
 func (e *Envoy) StopEnvoy() error {
-	// Close the access log server
-	if e.accessLogListener != nil {
-		e.accessLogListener.Close()
-		e.accessLogListener = nil
-	}
-	log.Info("Envoy: Stopping process ", e.cmd.Process.Pid)
-	e.rds.stop()
-	e.lds.stop()
-	err := e.admin.quit()
-	if err != nil {
-		log.WithError(err).Fatal("Envoy: Admin quit failed, killing process ", e.cmd.Process.Pid)
-		err := e.cmd.Process.Kill()
-		if err != nil {
-			log.WithError(err).Fatal("Envoy: Stopping failed")
-			return err
+	close(e.stopCh)
+
+	for {
```

Comment:
Unless you intended to do some other action in the default case, you don't need the loop or the select. reading from the channel (`<-e.errCh`)  will block indefinitely, waiting for something to be written to the channel, or if it closes. You might be able to do `return <-e.errCh` but the explicit `ok` handling is more clear.

Reply:
done